### PR TITLE
Fixes to CafeBots OnTurnProperty and BookTableDialog classes

### LINKED ISSUE #636: 51. Cafe Bot OnTurnPropert.cs error
## The bug
With the code written as is, line 55 was throwing an error saying that FirstOrDefault is not a method of JToken (Newtonsoft.Json.Linq.JEnumerable<Newtonsoft.Json.Linq.JToken>' does not contain a definition for 'FirstOrDefault'). 

## To Reproduce
1. Execute project
2. Use bot emulator
3. Enter an utterance that has a LUIS entity in it
4. place a line break at line 55 to catch error

## Expected behavior
It gives this error:
Newtonsoft.Json.Linq.JEnumerable<Newtonsoft.Json.Linq.JToken>' does not contain a definition for 'FirstOrDefault'

## Additional context
I was able to fix it with the following code:

value = (from val in (JArray)value
                              select val).FirstOrDefault();

[bug]

### PR DESCRIPTION
**Pull request by talon2king**
Fixes #636 
Line 55 of OnTurnProperty.cs file
Line 168 and 186 on BookTableDialog.cs file

## Proposed Changes

  - The fix in OnTurnProperty is to prevent an error being given for not being able to find the Method "FirstOrDefault" on JToken
  - The fixes in the BookTableDialog file fixes an issue where the bot was not returning an actual message to the emulator.

### CODE REVIEW

File: samples/csharp_dotnetcore/51.cafe-bot/Dialogs/BookTable/BookTableDialog.cs [hunk 1]
```diff
@@ -162,7 +162,12 @@ private async Task<DialogTurnResult> GetAllRequiredPropertiesAsync(WaterfallStep
                 // Start the prompt with the initial feedback based on update results.
                 var options = new PromptOptions()
                 {
-                    Prompt = new Activity(text: reservationResult.Outcome[0].Message),
+                    /* Had to change this to fix an issue with the bot emulator. The emulator was
```
Comment (JasonSowers) [inline samples/csharp_dotnetcore/51.cafe-bot/Dialogs/BookTable/BookTableDialog.cs]: please remove this comment

File: samples/csharp_dotnetcore/51.cafe-bot/Dialogs/BookTable/BookTableDialog.cs [hunk 2]
```diff
@@ -162,7 +162,12 @@ private async Task<DialogTurnResult> GetAllRequiredPropertiesAsync(WaterfallStep
...
+                       receiving a message from the bot that was missing the 'message' type in the JSON object.*/
+
+                       //TODO: Review
```
Comment (JasonSowers) [inline samples/csharp_dotnetcore/51.cafe-bot/Dialogs/BookTable/BookTableDialog.cs]: remove this todo

File: samples/csharp_dotnetcore/51.cafe-bot/Dialogs/BookTable/BookTableDialog.cs [hunk 3]
```diff
@@ -162,7 +162,12 @@ private async Task<DialogTurnResult> GetAllRequiredPropertiesAsync(WaterfallStep
                 // Start the prompt with the initial feedback based on update results.
                 var options = new PromptOptions()
                 {
-                    Prompt = new Activity(text: reservationResult.Outcome[0].Message),
+                    /* Had to change this to fix an issue with the bot emulator. The emulator was
+                       receiving a message from the bot that was missing the 'message' type in the JSON object.*/
+
+                       //TODO: Review 
+                    Prompt = MessageFactory.Text(reservationResult.Outcome[0].Message),
+                    //Prompt = new Activity(text: reservationResult.Outcome[0].Message),
```
Comment (JasonSowers) [inline samples/csharp_dotnetcore/51.cafe-bot/Dialogs/BookTable/BookTableDialog.cs]: remove the commented out code

File: samples/csharp_dotnetcore/51.cafe-bot/Dialogs/BookTable/BookTableDialog.cs [hunk 4]
```diff
@@ -176,7 +181,12 @@ private async Task<DialogTurnResult> GetAllRequiredPropertiesAsync(WaterfallStep
 
                 var options = new PromptOptions()
                 {
-                    Prompt = new Activity(text: reservationResult.NewReservation.GetMissingPropertyReadOut()),
+                    /* Had to change this to fix an issue with the bot emulator. The emulator was
```
Comment (JasonSowers) [inline samples/csharp_dotnetcore/51.cafe-bot/Dialogs/BookTable/BookTableDialog.cs]: please remove this comment

File: samples/csharp_dotnetcore/51.cafe-bot/Dialogs/BookTable/BookTableDialog.cs [hunk 5]
```diff
@@ -176,7 +181,12 @@ private async Task<DialogTurnResult> GetAllRequiredPropertiesAsync(WaterfallStep
...
+                       receiving a message from the bot that was missing the 'message' type in the JSON object.*/
+                       
+                       //TODO: Review
```
Comment (JasonSowers) [inline samples/csharp_dotnetcore/51.cafe-bot/Dialogs/BookTable/BookTableDialog.cs]: remove the todo

File: samples/csharp_dotnetcore/51.cafe-bot/Dialogs/BookTable/BookTableDialog.cs [hunk 6]
```diff
@@ -176,7 +181,12 @@ private async Task<DialogTurnResult> GetAllRequiredPropertiesAsync(WaterfallStep
 
                 var options = new PromptOptions()
                 {
-                    Prompt = new Activity(text: reservationResult.NewReservation.GetMissingPropertyReadOut()),
+                    /* Had to change this to fix an issue with the bot emulator. The emulator was
+                       receiving a message from the bot that was missing the 'message' type in the JSON object.*/
+                       
+                       //TODO: Review 
+                    Prompt = MessageFactory.Text(reservationResult.NewReservation.GetMissingPropertyReadOut()),
+                    //Prompt = new Activity(text: reservationResult.NewReservation.GetMissingPropertyReadOut()),
```
Comment (JasonSowers) [inline samples/csharp_dotnetcore/51.cafe-bot/Dialogs/BookTable/BookTableDialog.cs]: remove commented out code

File: samples/csharp_dotnetcore/51.cafe-bot/Dialogs/Shared/StateProperties/OnTurnProperty.cs [hunk 7]
```diff
@@ -52,7 +52,10 @@ public static OnTurnProperty FromLuisResults(RecognizerResult luisResults)
                 if (value is JArray)
                 {
                     // ConfirmList is nested arrays.
-                    value = value.Children().FirstOrDefault().FirstOrDefault();
+                    
+                    //TODO: Review fix:
```
Comment (JasonSowers) [inline samples/csharp_dotnetcore/51.cafe-bot/Dialogs/Shared/StateProperties/OnTurnProperty.cs]: remove todo

**Comment by talon2king** [comment]
@JasonSowers I was able to get the code cleaned up for you! Let me know if there's anything else that needs fixed.

**Comment by cleemullins** [comment]
@JasonSowers This looks good to me. Can you approve and merge?

**Pull request merged**